### PR TITLE
Replace internal reverse_postorder by a stable one

### PR DESCRIPTION
**Pull request by celinval**
The `reverse_postorder` function used before is internal to the compiler and reflects the internal body representation.

Besides that, I would like to introduce transformation on the top of SMIR body, which will break the 1:1 relationship between basic blocks from internal body and monomorphic body from StableMIR.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Comment by zhassan-aws** [comment]
Can you explain the motivation for this change? Is the compiler's `reverse_postorder` function unstable in some way?

**Comment by celinval** [comment]
> Can you explain the motivation for this change? Is the compiler's `reverse_postorder` function unstable in some way?

Totally. I updated the description. Let me know if that answers your question.

File: kani-compiler/src/codegen_cprover_gotoc/codegen/block.rs [hunk 1]
```diff
@@ -72,3 +73,25 @@ impl<'tcx> GotocCtx<'tcx> {
         }
     }
 }
+
```
Comment (zhassan-aws) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/block.rs:76]: Can you replicate the explanation in the PR's description here? Just want to make sure we don't wonder in the future why we're not using the version available from the compiler :)

**Review by zhassan-aws — APPROVED** [review]

File: kani-compiler/src/codegen_cprover_gotoc/codegen/block.rs (hunk 1 above)
Comment (celinval) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/block.rs:76]: sure

**Pull request merged**